### PR TITLE
Add loader for .graqhql files

### LINKED ISSUE #269: Add loader for graphq files
This is going to land in the next version of create react app, would be nice to have it here as well :)

https://github.com/facebook/create-react-app/pull/3909/files

### PR DESCRIPTION
**Pull request by patrick91**
Closes #269

See: https://github.com/facebook/create-react-app/pull/3909

### CODE REVIEW

**Comment by bahkostya** [comment]
Can somebody from maintainers look at this pull request Please? @DorianGrey, @wmonk 
Thanks in advance

File: packages/react-scripts/config/webpack.config.dev.js [hunk 1]
```diff
@@ -183,6 +183,11 @@ module.exports = {
               },
             ],
           },
+          // The GraphQL loader preprocesses GraphQL queries in .graphql files.
+          {
+            test: /\.(graphql)$/,
+            loader: 'graphql-tag/loader',
```
Comment (DorianGrey) [inline packages/react-scripts/config/webpack.config.dev.js]: Is there any particular reason that this loader is only registered in `dev`, but not in `prod` mode? (See the `webpack.config.prod.js`)

File: packages/react-scripts/template/README.md [hunk 2]
```diff
@@ -677,6 +678,34 @@ Please be advised that this is also a custom feature of Webpack.
 **It is not required for React** but many people enjoy it (and React Native uses a similar mechanism for images).<br>
 An alternative way of handling static assets is described in the next section.
 
+## Adding GraphQL files
+
+> Note: this feature is available with react-scripts@2.0.0 and higher.
```
Comment (DorianGrey) [inline packages/react-scripts/template/README.md]: Suppose this note is not required here, since this PR results in some kind of "pre-picking".

**Comment by simsim0709** [comment]
Hi, @patrick91 

Are you still working on this pr? If not, I would like to make a pr again, cuz I need it for my project.

Thank you

**Comment by patrick91** [comment]
I'll finish this in the weekend if that's ok :)

**Comment by simsim0709** [comment]
Yop. No problem. Thank you for this pr!! 😀

**Comment by eddiemoore** [comment]
Any progress on this?

**Comment by patrick91** [comment]
Sorry folks! I was travelling, I just pushed the changes, thanks for the nudges @eddiemoore @simsim0709 :)

@DorianGrey this should be fine to merge now, as soon as I fix the failure on travis ci :)

**Comment by patrick91** [comment]
Ok, I'm not sure why it fails, @DorianGrey can you tell me how to run the single failing test locally? I had troubles with docker unfortunately

**Comment by patrick91** [comment]
@DorianGrey yeah, I saw that, I was able to run the kitchensink locally but with no luck, it seems like it using an old version of the scripts, I even tried to delete the config and it didn't break, mmh, let me try again

**Comment by patrick91** [comment]
@DorianGrey it seems that we need to pass `--scripts-version=react-scripts-ts` to 
```
npx create-react-app --internal-testing-template="$root_path"/packages/react-scripts/fixtures/kitchensink test-kitchensink
```

so it will load the correct scripts, but the test fails still, with a different failure

```
/v/f/g/_/T/t/test-kitchensink (py3.6) REACT_APP_SHELL_ENV_MESSAGE=fromtheshell \
  NODE_PATH=src \
  PUBLIC_URL=http://www.example.org/spa/ \
  yarn build


yarn run v1.6.0
$ react-scripts-ts build
fs.js:987
  return binding.stat(pathModule._makeLong(path));
                 ^

Error: ENOENT: no such file or directory, stat '/private/var/folders/gy/_nycm7nx4zd45ldqdz5lzhk40000gn/T/tmp.hTn8c68y/test-kitchensink/tsconfig.json'
    at Error (native)
    at Object.fs.statSync (fs.js:987:18)
    at resolveConfigPath (/private/var/folders/gy/_nycm7nx4zd45ldqdz5lzhk40000gn/T/tmp.hTn8c68y/test-kitchensink/node_modules/tsconfig-paths/lib/tsconfig-loader.js:41:12)
    at loadSyncDefault (/private/var/folders/gy/_nycm7nx4zd45ldqdz5lzhk40000gn/T/tmp.hTn8c68y/test-kitchensink/node_modules/tsconfig-paths/lib/tsconfig-loader.js:19:22)
    at tsConfigLoader (/private/var/folders/gy/_nycm7nx4zd45ldqdz5lzhk40000gn/T/tmp.hTn8c68y/test-kitchensink/node_modules/tsconfig-paths/lib/tsconfig-loader.js:13:22)
    at configLoader (/private/var/folders/gy/_nycm7nx4zd45ldqdz5lzhk40000gn/T/tmp.hTn8c68y/test-kitchensink/node_modules/tsconfig-paths/lib/config-loader.js:27:22)
    at Object.loadConfig (/private/var/folders/gy/_nycm7nx4zd45ldqdz5lzhk40000gn/T/tmp.hTn8c68y/test-kitchensink/node_modules/tsconfig-paths/lib/config-loader.js:8:12)
    at new TsconfigPathsPlugin (/private/var/folders/gy/_nycm7nx4zd45ldqdz5lzhk40000gn/T/tmp.hTn8c68y/test-kitchensink/node_modules/tsconfig-paths-webpack-plugin/lib/plugin.js:20:42)
    at Object.<anonymous> (/private/var/folders/gy/_nycm7nx4zd45ldqdz5lzhk40000gn/T/tmp.hTn8c68y/test-kitchensink/node_modules/react-scripts-ts/config/webpack.config.prod.js:134:7)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
error Command failed with exit code 1.
```

**Comment by DorianGrey** [comment]
Hm... I'm not sure about adding this, since `--internal-testing-template` refers to a particular template that is used for the tests (in `packages/react-scripts/fixtures/kitchensink`). That particular template does **not** include any `typescript` related stuff.
That's why adding `--scripts-version=react-scripts-ts` does not work for that test: this parameter overwrites the template used for the test, thus there is no `tsconfig.json` as the build script expects.

**Comment by patrick91** [comment]
Yup, but without it we are testing the wrong scripts right? I'm happy to do another PR with the updated tests if that's ok :)

**Comment by DorianGrey** [comment]
The `--internal-testing-template` overwrites the template used during the `init` process on project generation, ignoring the one caused by the `--scripts-version` option. The resulting project contains this enforced template, but still utilizes the build configuration provided by the scripts package.

It might take a little more to set this up properly, i.e. modifying this particular template. PR is welcome, though.

**Comment by heresandyboy** [comment]
Hi, I have been trying to figure out how to use .graphql files in create-react-app-typescript. My investigations have led me to this PR - was support for loading .graphql ever implemented? Are there any workarounds available? My main reason for switching from .ts files using gql`` is so that I can easily use fragments.
